File: app/src/main/java/com/wzq/sample/data/source/http/service/DemoApiService.java
package com.wzq.sample.data.source.http.service;


import com.wzq.mvvmsmart.http.BaseResponse;
import com.wzq.sample.bean.DemoBean;

import io.reactivex.Observable;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Query;


public interface DemoApiService {

    @GET("action/apiv2/banner")
    Observable<BaseResponse<DemoBean>> demoGet(@Query("catalog") int pageNum);

    @FormUrlEncoded
    @POST("action/apiv2/banner")
    Observable<BaseResponse<DemoBean>> demoPost(@Field("catalog") String catalog);

    @GET("getJsonFile")
    Observable<BaseResponse<Object>> getJsonFile();
}

File: app/src/main/java/com/wzq/sample/data/source/http/service/DemoApiService.kt
package com.wzq.sample.data.source.http.service

import com.wzq.mvvmsmart.http.BaseResponse
import com.wzq.sample.bean.DemoBean
import io.reactivex.Observable
import retrofit2.http.*

/**
 * created 王志强 2020.04.30
 */
interface DemoApiService {
    @GET("action/apiv2/banner")
    fun demoGet(@Query("catalog") pageNum: Int): Observable<BaseResponse<DemoBean>>

    @FormUrlEncoded
    @POST("action/apiv2/banner")
    fun demoPost(@Field("catalog") catalog: String?): Observable<BaseResponse<DemoBean>>

    @get:GET("getJsonFile")
    val jsonFile: Observable<BaseResponse<Any>>

}